generators/ci-cd/templates/jenkins/Jenkinsfile.ejs

```diff
@@ -154,23 +154,16 @@ node {
 <%_ if (cicdIntegrations.includes('publishDocker')) { _%>
 
 <%= gitLabIndent %>    def dockerImage
-<%= gitLabIndent %>    stage('build docker') {
+<%= gitLabIndent %>    stage('publish docker') {
+<%= gitLabIndent %>        // A pre-requisite to this step is to setup authentication to the docker registry
     <%_ if (buildTool === 'maven') { _%>
-<%= gitLabIndent %>        sh "cp -R src/main/docker target/"
-<%= gitLabIndent %>        sh "cp target/*.jar target/docker/"
-<%= gitLabIndent %>        dockerImage = docker.build('<%= dockerRegistryOrganizationName %>/<%= baseName.toLowerCase() %>', 'target/docker')
+<%= gitLabIndent %>        // https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#authentication-methods
+<%= gitLabIndent %>        sh "./mvnw jib:build"
```

Comment:
I think jib doesn't use the docker daemon. So this could probably be done inside the build container without worrying about docker-in-docker

Reply:
This is a good point. I will try to change it.